extensions/amp-loader/0.1/amp-loader.js

```diff
@@ -0,0 +1,453 @@
+      if (logo.isDefault) {
+        // default logo is special because it fades away.
+        logoWrapper.classList.add('i-amphtml-new-loader-logo-default');
+      }
+      logoWrapper.appendChild(logo.svg);
+      this.svgRoot_.appendChild(logoWrapper);
+    }
+
+    this.svgRoot_.appendChild(spinner);
+  }
+
+  /**
+   * @param {string} color
+   */
+  getSpinner_(color) {
+    const svg = svgFor(this.element_);
+    const spinnerWrapper = svg`
+      <g class="i-amphtml-new-loader-spinner">
+    `;
+    for (let i = 0; i < 4; i++) {
+      const spinnerSegment = svg`
+        <circle class="i-amphtml-new-loader-spinner-segment" cx="60" cy="60">
+        </circle>
+      `;
+      spinnerSegment.setAttribute('stroke', color);
+      spinnerWrapper.appendChild(spinnerSegment);
+    }
+
+    return spinnerWrapper;
+  }
+
+  /**
+   * Adds the default or branded logo.
+   * @private
+   */
+  getLogo_() {
+    const customLogo = this.getCustomLogo_();
+    const useDefaultLogo = !customLogo;
+    const logo = customLogo || this.getDefaultLogo_();
+
+    // Ads always get the logo regardless of size
+    if (this.isAd_()) {
+      return logo;
+    }
+
+    // Small hosts do not get a logo
+    if (this.isSmall_()) {
+      return;
+    }
+
+    // If element requires a background shim but logo is the default logo,
+    // we don't show the logo.
+    if (this.requiresBackgroundShim_() && useDefaultLogo) {
+      return;
+    }
+
+    return {
+      svg: logo,
+      color: logo.getAttribute('fill') || DEFAULT_LOGO_SPINNER_COLOR,
+      isDefault: useDefaultLogo,
+    };
+  }
+
+  /**
+   * Add a gray default placeholder if there isn't a placeholder already and
+   * other special cases.
+   * @private
+   */
+  maybeAddDefaultPlaceholder_() {
+    const hasPlaceholder = !!this.element_.getPlaceholder();
+    const hasPoster = this.element_.hasAttribute('poster');
+    if (hasPlaceholder || hasPoster) {
+      return;
+    }
+
+    // Is it whitelisted for default placeholder?
+    const tagName = this.element_.tagName.toUpperCase();
+    if (
+      DEFAULT_PLACEHOLDER_WHITELIST_NONE_VIDEO[tagName] || // static white list
+      isIframeVideoPlayerComponent(tagName) // regex for various video players
+    ) {
+      const html = htmlFor(this.element_);
+      const defaultPlaceholder = html`
+        <div placeholder class="i-amphtml-default-placeholder"></div>
+      `;
+      this.element_.insertBefore(defaultPlaceholder, this.element_.lastChild);
+    }
+  }
+
+  /**
+   * Returns the custom logo for the element if there is one.
+   * @private
+   * @return {?Element}
+   */
+  getCustomLogo_() {
+    // Keeping the video logo here short term.
+    // This is because there is no single CSS for all players, there is
+    // video-interface but not all players implement it. Also the SVG is not
+    // that big.
+    // We need to move most of loaders code out of v0 anyway, see
+    // https://github.com/ampproject/amphtml/issues/23108.
```

Comment:
Is this comment still relevant after this PR?

Reply:
Updated the comment. I think we may eventually want to move this out still, but certainly less pressure now that it is async loaded.